### PR TITLE
Scripts for quick sources backend setup

### PR DESCRIPTION
**Pull request by lzap**
Based on @adiabramovitch 's work, I created those scripts which should help new users to get Sources backend up and running in minutes. I automated everything, so even someone with a fresh Fedora should get it running just by executing a script or two.

See the README.sources file for additional information. I would love someone to test this, I did my best but my Fedora was upgraded from let's say Fedora 19 so it's a tad different configuration than others :)

Cheers

### CODE REVIEW

File: scripts/sources.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export DATABASE_HOST=localhost
+export DATABASE_PORT=5432
+export DATABASE_NAME=sources_devel
+export DATABASE_USER=sources_user
+# Local users are considered trusted via pg_hba.conf but
+# password must be present otherwise sources won'b boot.
+export DATABASE_PASSWORD=trusted
+export CACHE_HOST=localhost
+export CACHE_PORT=6379
+export CACHE_PASSWORD=no_authentication
```
Comment (ezr-ondrej) [inline scripts/sources.conf:17]: I do not think these are used :thinking: I was looking for them in their code and it is not there...

File: scripts/sources.conf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+#
+# Do not edit this file, create sources.local.conf for overriding.
+#
+
+# Some random port because 8000 is typically already used.
+export PORT=8131
```
Comment (ezr-ondrej) [inline scripts/sources.conf:7]: can we also set `METRICS_PORT`? They have 9000 as well as we do and it could collide, right?

**Review by ezr-ondrej — COMMENTED** [review]
nice, few nitpicks, but so minor I'm okay merging as is :shrug:

File: scripts/sources.conf (hunk 1 above)
Comment (lzap) [reply scripts/sources.conf:17]: @adiabramovitch can you comment? You had it there so perhaps there is a reason.

File: scripts/sources.conf (hunk 2 above)
Comment (lzap) [reply scripts/sources.conf:7]: Good point, added.

File: scripts/sources.conf (hunk 1 above)
Comment (adiabramovitch) [reply scripts/sources.conf:18]: I don't recall, If you didn't need those, maybe they are redundant.

**Comment by adiabramovitch** [comment]
Maybe we should add the org id to the account number that was seeded to the DB. (In order to authenticate both sources and provisioning)

**Comment by ezr-ondrej** [comment]
> Maybe we should add the org id to the account number that was seeded to the DB. (In order to authenticate both sources and provisioning)

That's already in our `dev_small`, right?

**Review by ezr-ondrej — APPROVED** [review]
Duplication is fine I guess, I'd get it in, it works and breaks nothing :)
@adiabramovitch anything to add?

File: scripts/sources.conf (hunk 1 above)
Comment (lzap) [reply scripts/sources.conf:18]: Ok let's drop it then.

**Comment by lzap** [comment]
Dropped the tree REDIS unused variables.

I am not sure how you want me to add account number, the poppullator does not support it:

```
$ ag Getenv
config/parser.go
83:	logLevel := os.Getenv("LOG_LEVEL")
91:	sourcesHost := os.Getenv("SOURCES_API_HOST")
97:	sourcesPort := os.Getenv("SOURCES_API_PORT")
107:	concurrentRequests := os.Getenv("CONCURRENT_REQUESTS")
125:	numberTenants := os.Getenv("NUMBER_OF_TENANTS")
167:	sourcesPerTenant := os.Getenv("SOURCES_PER_TENANT")
180:	rhcConnectionsPerTenant := os.Getenv("RHC_CONNECTIONS_PER_TENANT")
193:	applicationsPerSource := os.Getenv("APPLICATIONS_PER_SOURCE")
206:	endpointsPerSource := os.Getenv("ENDPOINTS_PER_SOURCE")
218:	authenticationsPerResource := os.Getenv("AUTHENTICATIONS_PER_RESOURCE")
```

**Comment by lzap** [comment]
Let me take a look on the seeder application to see how they actually create tenants. We can probably create our own one via curl too.

**Comment by adiabramovitch** [comment]
> > because they are saving only the account number and not the org id, and without adding this we can authenticate both of the services.
> 
> Ah! got it!
> 
> > Let me take a look on the seeder application to see how they actually create tenants. We can probably create our own one via curl too.
> 
> Could we just have a DB query adding static org_id into the db? :) I'd not worry about it too much to make it pretty, org_id should go away in a year anyway :)

I might be mistaken. 
I asked @lzap  to try a simple curl request through our service to check if it is authenticating both services, I think the seeding of the x-rh-identity occurs automatically.

**Comment by lzap** [comment]
Yep I can confirm, a simple curl that lists something automatically adds account number into the tenants table. It ignores org id tho, I think this is a bug (org id has only been added recently). Example:

```
curl --location --request GET 'http://localhost:8131/api/sources/v3.1/authentications' --header 'x-rh-identity: eyJpZGVudGl0eSI6IHsidHlwZSI6ICJVc2VyIiwgImFjY291bnRfbnVtYmVyIjoiNjA4OTcxOSIsICJpbnRlcm5hbCI6eyJvcmdfaWQiOiIwMDAwMTMifX19Cg==' | json_pp
```

**Comment by lzap** [comment]
Just for the record, I think we can take the same approach then. When we encounter a request with account/org ids, we would find or create the record in our accounts database. Do we want to create a ticket for that @ezr-ondrej ?

**Comment by lzap** [comment]
Created: https://issues.redhat.com/browse/HMSPROV-106

**Review by ezr-ondrej — APPROVED** [review]
While testing I found it bit challanging to having to swich directories to `/scripts` for this to work properly, but that's the only caveat, otherwise it works well, I don't want to block this on that, it is already quite useful as-is :)

**Comment by ezr-ondrej** [comment]
Thanks @lzap, great job! :+1:

**Pull request merged**